src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -268,11 +268,38 @@ protected BoundExpression LowerTest(BoundDagTest test)
                         Debug.Assert(!input.Type.IsNullableType());
                         return MakeEqual(_localRewriter.MakeLiteral(d.Syntax, d.Value, input.Type), input);
 
+                    case BoundDagRelationalTest d:
+                        Debug.Assert(!input.Type.IsNullableType());
+                        Debug.Assert(input.Type.TypeKind == TypeKind.Struct);
+                        return MakeRelational(input, d.OperatorKind, _localRewriter.MakeLiteral(d.Syntax, d.Value, input.Type));
+
                     default:
                         throw ExceptionUtilities.UnexpectedValue(test);
                 }
             }
 
+            private BoundExpression MakeRelational(BoundExpression input, BinaryOperatorKind operatorKind, BoundExpression literal)
+            {
+                TypeSymbol comparisonType = input.Type;
+                if (operatorKind.OperandTypes() == BinaryOperatorKind.Int && input.Type.SpecialType != SpecialType.System_Int32)
+                {
+                    // Promote operands to int before comparison for byte, sbyte, short, ushort
```

Comment:
>// Promote operands to int before comparison for byte, sbyte, short, ushort [](start = 20, length = 75)

I didn't understand why

Reply:
Because the language does not have `bool operator <(byte, byte);`  In order to accomplish that the operands are promoted to `int`.

---
In reply to: [381012334](https://github.com/dotnet/roslyn/pull/41503#discussion_r381012334) [](ancestors = 381012334)